src/query/expression/src/types/decimal.rs

```diff
@@ -725,6 +765,32 @@ impl Decimal for i256 {
         i256::from(value.into())
     }
 
+    fn from_bigint(value: BigInt) -> Option<Self> {
+        let mut ret: u256 = u256::ZERO;
+        let mut bits = 0;
+
+        for i in value.iter_u64_digits() {
+            if bits >= 256 {
+                return None;
+            }
+            ret |= u256::from(i) << bits;
+            bits += 64;
```

Comment:
least significant digit first.
the last i may only used a part of 64 bits?

Reply:
if `bits` is larger than 256, it's overflow, so return none.

This method is powered by https://docs.rs/num-bigint/latest/src/num_bigint/bigint/convert.rs.html#86-92